src/patcher/java/net/minecraftforge/gradle/patcher/PatcherPlugin.java

```diff
@@ -85,6 +86,7 @@ public void apply(@Nonnull Project project) {
         EnvironmentChecks.checkEnvironment(project);
 
         final PatcherExtension extension = project.getExtensions().create(PatcherExtension.class, PatcherExtension.EXTENSION_NAME, PatcherExtension.class, project);
+        project.getExtensions().create(LegacyExtension.EXTENSION_NAME, LegacyExtension.class);
```

Comment:
You should probably apply this after the channel providers just for consistency with userdev.